trieste/acquisition/rule.py

```diff
@@ -1103,6 +1115,380 @@ def state_func(
+        models: Mapping[Tag, ProbabilisticModelType],
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Reinitialise subspaces if necessary.
+        Get a mask of subspaces that need to be reinitialised using an abstract method.
+        Reinitialise individual subpaces by calling the method of the UpdateableSearchSpace class.
+        """
+        mask = self.get_reinitialise_subspaces_mask(subspaces, models, datasets)
+        for ix, subspace in enumerate(subspaces):
+            if mask[ix]:
+                subspace.reinitialise(models, datasets)
+
+    @abstractmethod
+    def get_reinitialise_subspaces_mask(
+        self,
+        subspaces: Sequence[UpdateableSearchSpaceType],
+        models: Mapping[Tag, ProbabilisticModelType],
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> TensorType:
+        """Get mask for subspaces that need to be reinitialised."""
+        ...
+
+
+class TrustRegionBox(Box, UpdateableSearchSpace):
+    """An updateable box search space for use with trust region acquisition rules."""
+
+    def __init__(
+        self,
+        global_search_space: SearchSpace,
+        beta: float = 0.7,
+        kappa: float = 1e-4,
+        min_eps: float = 1e-2,
+    ):
+        """
+        Calculates the bounds of the box from the location/centre and global bounds.
+
+        :param global_search_space: The global search space this search space lives in.
+        """
+
+        self._global_search_space = global_search_space
+        self._beta = beta
+        self._kappa = kappa
+        self._min_eps = min_eps
+        self._initialised = False
+
+        super().__init__(global_search_space.lower, global_search_space.upper)
+
+    @property
+    def global_search_space(self) -> SearchSpace:
+        """The global search space this search space lives in."""
+        return self._global_search_space
+
+    @property
+    def location(self) -> TensorType:
+        """The location of the search space."""
+        return self._location
+
+    @location.setter
+    def location(self, location: TensorType) -> None:
+        """Set the location of the search space."""
+        self._location = location
+
+    def reinitialise(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Reinitialise the box."""
+        _, dataset = self.get_single_model_and_dataset(models, datasets)
+
+        self.location = tf.squeeze(self.global_search_space.sample(1), axis=0)
+
+        global_lower = self.global_search_space.lower
+        global_upper = self.global_search_space.upper
+
+        if self._initialised:
+            scale = 0.1
+        else:
+            scale = 0.5
+            self._initialised = True
+
+        self._eps = scale * (global_upper - global_lower) / (5.0 ** (1.0 / global_lower.shape[-1]))
```

Comment:
I'll add the method. However, I have a related question.

Is the concrete `TrustRegionBox` class generally useful and worth keeping directly in Trieste (instead of e.g. a notebook)? My current intention was that the users would inherit from the abstract classes to implement their algorithms; and `TrustRegionBox` is just one implementation of an example algorithm.